File: ios/Classes/UrlClickHandler.swift
import Foundation
import Flutter

class UrlClickHandler: NSObject, FlutterStreamHandler{
    var sink: FlutterEventSink?

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }

    @objc func buttonClicked(url: String?) {
        guard let sink = sink else { return }
        let dictionary = ["URL": url]
        sink(dictionary)
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

File: android/src/main/kotlin/io/verloop/verloop_flutter_sdk/UrlClickHandler.kt
package io.verloop.verloop_flutter_sdk

import android.os.Handler
import android.os.Looper
import io.flutter.plugin.common.EventChannel


class UrlClickHandler : EventChannel.StreamHandler {
    private val uiThreadHandler: Handler = Handler(Looper.getMainLooper())
    var sink: EventChannel.EventSink? = null

    fun urlClicked(url: String?) {
        uiThreadHandler.post {
            sink?.success(
                mapOf(
                    "URL" to url,
                )
            )
        }
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }
}